PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/appservice/TunnelProxy.java

```diff
@@ -0,0 +1,100 @@
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.toolkit.lib.appservice;
+
+import com.google.common.base.Joiner;
+import com.jcraft.jsch.*;
+import com.microsoft.azure.management.appservice.PublishingProfile;
+import com.microsoft.azure.management.appservice.WebAppBase;
+import com.microsoft.azure.toolkit.lib.common.utils.WebSocketSSLProxy;
+import org.apache.commons.io.IOUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+public class TunnelProxy {
+    private static final Logger logger = Logger.getLogger(TunnelProxy.class.getName());
+    private static final String LOCALHOST = "localhost";
+    private static final String DEFAULT_SSH_USERNAME = "root";
+    private static final String DEFAULT_SSH_PASSWORD = "Docker!";
+    private WebAppBase webAppBase;
+    private WebSocketSSLProxy wssProxy;
+
+    public TunnelProxy(@NotNull WebAppBase webAppBase) {
+        this.webAppBase = webAppBase;
+        reset();
+    }
+
+    public void reset() {
+        String host = webAppBase.defaultHostName().toLowerCase().replace("http://", "").replace("https://", "");
+        String[] parts = host.split("\\.", 2);
+        host = Joiner.on('.').join(parts[0], "scm", parts[1]);
+        PublishingProfile publishingProfile = webAppBase.getPublishingProfile();
+        wssProxy = new WebSocketSSLProxy(String.format("wss://%s/AppServiceTunnel/Tunnel.ashx", host),
+                                         publishingProfile.gitUsername(), publishingProfile.gitPassword());
+    }
+
+    public void close() {
+        if (Objects.nonNull(wssProxy)) {
+            wssProxy.close();
+            wssProxy = null;
+        }
+    }
+
+    public String executeCommandViaSSH(String cmd) throws IOException {
+        if (Objects.isNull(wssProxy)) {
+            reset();
+        }
+        wssProxy.start();
```

Comment:
why start on every execution?
I can see `start` but no `close`

Reply:
1. wssProxy is not for one socket, we need to open new ssh connections at every command
2.  TunnelProxy has `close` method

Reply:
add close at the end

Reply:
seems you miss `close` at the end of this method.